macros/compareasm.swb

```diff
@@ -0,0 +1,72 @@
+' COMPARE SOLIDWORKS ASSEMBLIES
+' COPYRIGHT (C) 2021 HENRY FRANKS
+
+Option Explicit
+
+Declare PtrSafe Function lstrcpyn_long_string Lib "kernel32" Alias "lstrcpynA" (ByVal DestString As String, ByVal SourcePointer As Long, ByVal MaxLen As Long) As Long
+Declare PtrSafe Function lstrlen_long Lib "kernel32" Alias "lstrlenA" (ByVal SourcePointer As Long) As Long
+Declare PtrSafe Function GetCommandLine_long Lib "kernel32" Alias "GetCommandLineA" () As Long
+
+Sub main()
+
+    Dim Args()         As String
+    Dim swApp          As Object
+    Dim swUtil         As Object
+    Dim swUtilCompGeom As Object
+    Dim longStatus     As Long
+
+    Set swApp = CreateObject("SldWorks.Application")
+    Set swUtil = swApp.GetAddInObject("Utilities.UtilitiesApp")
+
+    If Not swUtil Is Nothing Then
+        Set swUtilCompGeom = swUtil.GetToolInterface(2, longStatus)
+        Args = Split(GetCommandLine, " /*")
+
+        longStatus = swUtilCompGeom.CompareGeometry(Args(1), Args(2), 0, 0, "")
+        longStatus = swUtilCompGeom.Close()
+
+        If longStatus <> 0 Then
+            MsgBox "Comparison failed with exit code " + Str(longStatus) _
+                   + ". Please open an issue at " _
+                   + "https://github.com/tim-clifford/sldworks-git-tools"
+        End
+    Else
+        MsgBox "Cannot open comparison - please make sure Solidworks Utilities is enabled at startup"
+    End If
```

Comment:
Is it possible for `CompareGeometry` to fail but `Close` to succeed? Should we duplicate the exit code check between each call?

Reply:
ugh selected too many lines and can't change it